api/subscriptions/message_cache.go

```diff
@@ -0,0 +1,66 @@
+// Copyright (c) 2024 The VeChainThor developers
+//
+// Distributed under the GNU Lesser General Public License v3.0 software license, see the accompanying
+// file LICENSE or <https://www.gnu.org/licenses/lgpl-3.0.html>
+
+package subscriptions
+
+import (
+	"fmt"
+	"sync"
+
+	lru "github.com/hashicorp/golang-lru"
+	"github.com/vechain/thor/v2/thor"
+)
+
+// messageCache is a generic cache that stores messages of any type.
+type messageCache[T any] struct {
+	cache *lru.Cache
```

Comment:
In this scenario, should use lock free versioned `github.com/hashicorp/golang-lru/simplelru`